helper/logging/logging_http_transport_test.go

```diff
@@ -0,0 +1,233 @@
+package logging_test
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-log/tflogtest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/logging"
+)
+
+func TestNewLoggingHTTPTransport(t *testing.T) {
+	ctx, loggerOutput := setupRootLogger()
+
+	transport := logging.NewLoggingHTTPTransport(http.DefaultTransport)
+	client := http.Client{
+		Transport: transport,
+		Timeout:   10 * time.Second,
+	}
+
+	reqBody := `An example
+		multiline
+		request body`
+	req, _ := http.NewRequest("GET", "https://www.terraform.io", bytes.NewBufferString(reqBody))
+	res, err := client.Do(req.WithContext(ctx))
+	if err != nil {
+		t.Fatalf("request failed: %v", err)
+	}
+	defer res.Body.Close()
+
+	entries, err := tflogtest.MultilineJSONDecode(loggerOutput)
+	if err != nil {
+		t.Fatalf("log outtput parsing failed: %v", err)
+	}
+
+	if len(entries) != 2 {
+		t.Fatalf("unexpected amount of logs produced; expected 2, got %d", len(entries))
+	}
+
+	reqEntry := entries[0]
+	if diff := cmp.Diff(reqEntry, map[string]interface{}{
+		"@level":              "debug",
+		"@message":            "An example multiline request body",
```

Comment:
I wonder if instead of putting the body into the message field, we should instead put it into its own `tf_http_req_body` (or whatever) field and set the message to something like "Sending HTTP request"?

Reply:
I think we talked in zoom briefly about this. The only argument I could find for this, was that the "log masking" can be applied to a portion of the `@message` this way.
If we make the body a field, it's a "all or nothing": either they mask the whole value, or not.

I can go either way. I just feel that, in the case of masking, it would be "too blunt". 

What's your preference?

Reply:
Aye, yes given the current filtering functionality, that does make this a little less developer friendly. 🙁 Maybe it is worth getting this developer-facing problem sorted before potentially changing it later? Flipping this problem around, being able to also have the option to redact the entire body via field key would be kinda nice (and is generally a feature for HTTP loggers)!

Regardless of where we land with this particular functionality, I think adding the ability to mask _portions_ of field values makes sense. Whether that masking applies to "any key" or "this specific key" is probably an important distinction, sigh. I'll create an upstream feature request issue for that. If distinguishing between message masking and field masking is too much of an ask, then maybe having a "just mask all the things" thing is appropriate.

Reply:
I think it's unlikely that something that a user explicitly wants masked in a message, would be ok to leave plan in a field (and viceversa).

I think the correct thing to do would be to apply the same "find and replace" we do to messages, to the content of each key in the fields.

Reply:
All right, here is a PR to address the missing feature in terraform-plugin-log: https://github.com/hashicorp/terraform-plugin-log/pull/87

Reply:
Code that implements this, is now pushed.

The version of `terraform-plugin-log` has to be released before we can merge though.